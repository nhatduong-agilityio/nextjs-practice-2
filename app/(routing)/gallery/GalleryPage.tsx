'use client'

import { Flex, Grid, GridItem, Heading, Stack } from '@chakra-ui/react'
import React, { Fragment, memo } from 'react'
import NextLink from 'next/link'
import Image from '@components/Common/Image'
import { switcherGridItemProps } from '@helper/gridGallery'
import type { TArticles } from '@matched-types/articles'
import { PAGE_URL } from '@constants/routes'

interface IProps {
  articles: TArticles
}

const GalleryPage = memo(({ articles }: IProps) => (
  <Flex w='full' flexDir='column' p={['10px 20px', '50px 100px', '50px 100px']} pt='30px'>
    <Stack
      flexDir='row'
      justifyContent='flex-start'
      alignItems='center'
      pb={['25px', '50px', '50px']}
    >
      <Heading variant={['extraBoldMD', 'extraBoldXL']} lineHeight='inherit'>
        Interior Designer
      </Heading>
    </Stack>
    <Grid
      h='100vh'
      templateRows={['repeat(5, 1fr)', 'repeat(7, 1fr)']}
      templateColumns={['repeat(4, 1fr)']}
      gap={['10px', '50px', '50px']}
    >
      {articles.map((item, index) => (
        <Fragment key={item.id}>
          {/** function helper render gird item props by index of array */}
          <GridItem {...switcherGridItemProps(index)}>
            <NextLink href={`${PAGE_URL.GALLERY.URL}/${item.id}`} passHref>
              <Image alt={item.imageAlt} src={item.image} width='full' height='full' />
            </NextLink>
          </GridItem>
        </Fragment>
      ))}
    </Grid>
  </Flex>
))

GalleryPage.displayName = 'GalleryPage'

export default GalleryPage
