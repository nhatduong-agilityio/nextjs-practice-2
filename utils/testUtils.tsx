import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
