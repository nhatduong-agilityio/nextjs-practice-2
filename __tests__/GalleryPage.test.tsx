import '@testing-library/jest-dom'

import type { RenderResult } from '@utils/testUtils'
import { cleanup, render, waitFor } from '@utils/testUtils'

import { ArticlesMock } from '@mock/dataMock'
import GalleryPage from '../app/(routing)/gallery/GalleryPage'

jest.mock('next/navigation', () => ({
  useSearchParams: jest.fn(),
}))

describe('GalleryPage', () => {
  global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve(),
    }),
  ) as jest.Mock

  const props = { articles: ArticlesMock }

  let wrapper: RenderResult

  const setup = async () => {
    const view = render(<GalleryPage {...props} />)

    await waitFor(() => {
      wrapper = view
    })
  }

  beforeEach(() => {
    cleanup()

    setup()
  })

  it('Should GalleryPage match snapshot', async () => {
    const { container } = wrapper

    expect(container).toMatchSnapshot()
  })
})
