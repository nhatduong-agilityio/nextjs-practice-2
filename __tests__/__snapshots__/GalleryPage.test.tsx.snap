// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryPage Should GalleryPage match snapshot 1`] = `
<div>
  <div
    class="css-1gw30fa"
  >
    <div
      class="chakra-stack css-1yykub2"
    >
      <h2
        class="chakra-heading css-1k4oj01"
      >
        Interior Designer
      </h2>
    </div>
    <div
      class="css-121qhnu"
    >
      <div
        class="css-1lz6fgg"
      >
        <a
          href="/gallery/a69c31b7-948e-40a8-b900-597931d2cb3c"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture chair and table office"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-2bei10"
      >
        <a
          href="/gallery/72a207a4-fb43-4e71-8100-f43a156b68a3"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture desk office"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-15iqidc"
      >
        <a
          href="/gallery/0321c4bf-ef9b-49ea-8da9-34ba1b754336"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture sofa"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-nrxmu7"
      >
        <a
          href="/gallery/3333b8a9-8c38-4e8a-970a-28aac53f0e51"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture chair and table office2"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-1xz837u"
      >
        <a
          href="/gallery/2798034b-3e3b-4c0e-a9e7-29e5d769b5c9"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture desk office2"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-plg8k9"
      >
        <a
          href="/gallery/a568cf09-0d88-48ec-bbd9-624eac45e50c"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture sofa2"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-4gox30"
      >
        <a
          href="/gallery/b0334801-3cd4-44ea-87d3-c0587cf684bc"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture chair and table office3"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-18i2y6i"
      >
        <a
          href="/gallery/64cd37f2-9be3-4189-a7d5-fe120f78e016"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture desk office3"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-ku5lu7"
      >
        <a
          href="/gallery/3e4caf03-cc19-4197-aa2b-286adcee6dc0"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture sofa3"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-hfwe92"
      >
        <a
          href="/gallery/1730aba5-dd0d-4fc3-b983-2f1a30c7b7b8"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture chair and table office4"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-hfwe92"
      >
        <a
          href="/gallery/4291ed23-cb5a-4d72-b06d-4eb37ac42dc2"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture desk office4"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-1803f6"
      >
        <a
          href="/gallery/033a5be2-2656-450e-9ed9-fd55666cfb0a"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture sofa4"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
